### PR TITLE
libretro-snes9x2002: Add loading hacks from snesadvance.dat file

### PR DESCRIPTION
**Pull request by tiopex**
Add loading hacks from snesadvance.dat file function from drPocketSnes emu

### CODE REVIEW

**Comment by Apaczer** [comment]
AFAIR the speedhacks may brake/reduce accuracy on some parts (by skipping some CPU cycles), so you would likely want to make it optional. 
BTW there is dynarec code in snes9x2002 which may help with snes9x2005 (see https://git.crowdedwood.com/snes9x2005/log/?h=performance)

**Comment by tiopex** [comment]
That's the plan to add an option to the ra settings

**Comment by Apaczer** [comment]
You should merge here only then and if succed try to push it upstream, some ppl may like it. Considering it hasn't got any signiffact updates in years (last from @jSTE0, cheers!).

**Comment by jSTE0** [comment]
Previously discussed in TriForceX/MiyooCFW#479 in the context of snes9x4d and was known to break some games as noted in TriForceX/MiyooCFW#380.

File: package/libretro-snes9x2002/0002-Add-snesadvance.patch [hunk 1]
```diff
@@ -0,0 +1,770 @@
...
++int os9x_findhacks(int game_crc32){
++	int i=0,j;
++	int _crc32;	
++	char c;
++	char str[256];
++	unsigned int size_snesadvance;
++	unsigned char *snesadvance;
++	FILE *f;
++	
++	sprintf(str,"%s/snesadvance.dat","/mnt/.retroarch/system/snes9x2002");
```
Comment (Apaczer) [inline package/libretro-snes9x2002/0002-Add-snesadvance.patch]: if you want to specify system dir use maybe ifdef \_\_LIBRETRO\_\_ and just call RETRO_ENVIRONMENT_GET_SYSTEM_DIRECTORY to make it look cleaner

**Review by Apaczer — CHANGES_REQUESTED** [review]
seems to work 👍  ("Batman Returs" brakes at startup), but you need full core restart (close and launch again).
Anyway I'm going offline for a month (✈️ ), so this will be my last request.

**Review by Apaczer — APPROVED** [review]
Normally this shouldn't be enabled by default, so I will leave merging to you 😏

**Comment by tiopex** [comment]
Done

**Comment by Apaczer** [comment]
GJ on this one

**Pull request merged**